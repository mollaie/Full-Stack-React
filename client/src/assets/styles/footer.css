.footer {
  position: fixed;
  bottom: 0;
  right: 0;
  left: 0;
  padding: 0.5rem;
  min-height: 4.5rem;
  background-color: #ffffff;
  display: flex;
  width: 100%;
  flex-direction: row;
  align-items: center;
  justify-content: flex-end;
  border-top: 1px solid #cacaca;
}

.create-action-bar {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  flex: 1;
}

button {
  width: 45%;
}
