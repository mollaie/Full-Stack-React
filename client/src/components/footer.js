import React, { useState, useEffect } from "react";
import { useNavigate } from "react-router-dom";

import "../assets/styles/footer.css";

const Footer = (props) => {
  const navigate = useNavigate();

  const createReservationHandler = (e) => {
    console.log(e);
    e.preventDefault();

    navigate("/Create");
  };

  const cancelReservation = (e) => {
    e.preventDefault();

    navigate("/");
  };

  const submitReservation = (e) => {
    e.preventDefault();

    navigate("/");
  };

  const DefaultFooter = () => {
    return (
      <div className="footer">
        <button className="btn btn-primary" onClick={createReservationHandler}>
          Create Reservation
        </button>
      </div>
    );
  };

  const CreateFooter = () => {
    return (
      <div className="footer">
        <div className="create-action-bar">
          <button className="btn btn-secondary" onClick={cancelReservation}>
            Cancel
          </button>
          <button className="btn btn-success" onClick={submitReservation}>
            Create
          </button>
        </div>
      </div>
    );
  };

  const [isCreate, setIsCreate] = useState(false);

  useEffect(() => {
    setIsCreate(props.isCreate);
  }, [props.isCreate]);

  return !isCreate ? <DefaultFooter /> : <CreateFooter />;
};

export default Footer;
