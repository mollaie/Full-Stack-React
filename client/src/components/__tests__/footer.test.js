import React from "react";
import renderer from "react-test-renderer";
import footer from "../footer";

describe("footer test", () => {
  it("footer should match snapshot", () => {
    const component = renderer.create(<footer />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
