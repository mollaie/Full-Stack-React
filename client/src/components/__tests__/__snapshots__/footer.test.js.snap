// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer test footer should match snapshot 1`] = `<footer />`;
